src/hooks/screen/useAnonymousChatScreenHook.ts

```diff
@@ -0,0 +1,99 @@
+// eslint-disable-next-line import/no-unresolved
+import 'react-native-get-random-values';
+
+import * as React from 'react';
+import SimpleToast from 'react-native-simple-toast';
+import {v4 as uuid} from 'uuid';
+
+import AnonymousMessageRepo from '../../service/repo/anonymousMessageRepo';
+import ChatSchema from '../../database/schema/ChatSchema';
+import UseAnonymousChatScreenHook from '../../../types/hooks/screens/useAnonymousChatScreenHook.types';
+import useChatUtilsHook from '../core/chat/useChatUtilsHook';
+import useLocalDatabaseHook from '../../database/hooks/useLocalDatabaseHook';
+import {getAnonymousUserId} from '../../utils/users';
+import {getUserId} from '../../utils/token';
+import {randomString} from '../../utils/string/StringUtils';
+
+function useAnonymousChatScreenHook(): UseAnonymousChatScreenHook {
+  const {localDb, chat, refresh} = useLocalDatabaseHook();
+  const {selectedChannel, goBackFromChatScreen, goToChatInfoScreen} = useChatUtilsHook();
+
+  const [chats, setChats] = React.useState<ChatSchema[]>([]);
+
+  const initChatData = async () => {
+    if (!localDb && !selectedChannel) return;
+    const myUserId = await getUserId();
+    const myAnonymousId = await getAnonymousUserId();
+    const data = (await ChatSchema.getAll(
+      localDb,
+      selectedChannel?.id,
+      myUserId,
+      myAnonymousId
+    )) as ChatSchema[];
+    setChats(data);
+  };
+
+  const sendChat = async (
+    message: string = randomString(20),
+    iteration = 0,
+    sendingChatSchema: ChatSchema = null
+  ) => {
+    if (iteration > 5) {
+      SimpleToast.show("Can't send message, please check your connection");
+      return;
+    }
+
+    let currentChatSchema = sendingChatSchema;
+
+    const myAnonymousId = await getAnonymousUserId();
+    try {
+      const randomId = uuid();
+
+      if (currentChatSchema === null) {
+        currentChatSchema = await ChatSchema.generateSendingChat(
+          randomId,
+          myAnonymousId,
+          selectedChannel?.id,
+          message,
+          localDb
+        );
+
+        await currentChatSchema.save(localDb);
+        refresh('chat');
+        refresh('channelList');
+      }
+
+      const response = await AnonymousMessageRepo.sendAnonymousMessage(
+        selectedChannel?.id,
+        message
+      );
+
+      await currentChatSchema.updateChatSentStatus(localDb, response);
+      refresh('chat');
+      refresh('channelList');
+    } catch (e) {
+      console.log(e);
+      if (e?.response?.data?.status === 'Channel is blocked') return;
+
+      setTimeout(() => {
+        sendChat(message, iteration + 1, currentChatSchema).catch((sendChatError) => {
+          console.log(sendChatError);
+        });
+      }, 1000);
+    }
+  };
+
+  React.useEffect(() => {
+    initChatData();
+  }, [localDb, chat, selectedChannel]);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This `useEffect` hook will run every time `localDb`, `chat`, or `selectedChannel` changes. If these values change frequently, it might lead to performance issues as `initChatData` is an async function that performs database operations. Consider adding a loading state to prevent unnecessary calls.
```diff
-  React.useEffect(() => {
-    initChatData();
-  }, [localDb, chat, selectedChannel]);
+  const [loading, setLoading] = React.useState(false);
+
+  React.useEffect(() => {
+    if (!loading) {
+      setLoading(true);
+      initChatData().finally(() => setLoading(false));
+    }
+  }, [localDb, chat, selectedChannel]);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->